docs/workflow/testing/libraries/filtering-tests.md

```diff
@@ -176,3 +234,67 @@ dotnet build <csproj_file> /t:Test /p:TargetOS=windows
 ```sh
 dotnet build <csproj_file> /t:Test /p:TargetOS=OSX /p:WithCategories="OuterLoop;failing""
 ```
+
+## SkipOnCoreClrAttribute
+This attribute is used to disable a test under specific conditions, only when run with CoreCLR (it doesn't affect tests run with Mono). Typically, this is when there is a failure in a particular test run configuration, such as under GCStress or JitStress.
+
+This attribute can be applied either to a test class (will disable all the tests in that class) or to a test method. It allows multiple usages on the same member.
+
+This attribute is defined [here](https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.XUnitExtensions/src/Attributes/SkipOnCoreClrAttribute.cs).
+
+**Disable for all platforms and all target frameworks:**
+
+```cs
+[SkipOnCoreClr(string reason)]
+```
+Example:
+```cs
+[SkipOnCoreClr("CoreCLR does track thread specific JIT information")]
+```
+
+**Disable for specific platform:**
+
+```cs
+[SkipOnCoreClr(string reason, TestPlatforms testPlatforms)]
+```
+Example:
+```cs
+[SkipOnCoreClr("Long running tests: https://github.com/dotnet/runtime/issues/11980", TestPlatforms.Linux)
+```
+
+**Disable for specific test mode:**
+
+A test mode is a run configuration, like JitStress, JitStressRegs, JitMinOpts, TailcallStress, GCStress.
+`RuntimeTestModes` is defined [here](https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.XUnitExtensions/src/RuntimeTestModes.cs).
+
+```cs
+[SkipOnCoreClr(string reason, RuntimeTestModes testMode)]
+```
+Examples:
+```cs
+[SkipOnCoreClr("https://github.com/dotnet/runtime/issues/60240", RuntimeTestModes.JitStressRegs)]
+[SkipOnCoreClr("Long running tests: https://github.com/dotnet/runtime/issues/10680", RuntimeTestModes.JitMinOpts)]
+```
+
+**Disable for specific runtime configuration:**
+
+A runtime configuration is a the build of the runtime: Debug, Checked, Release.
+`RuntimeConfiguration` is defined [here](https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.XUnitExtensions/src/RuntimeConfiguration.cs).
+
+```cs
+[SkipOnCoreClr(string reason, RuntimeConfiguration runtimeConfigurations)]
+```
+Example:
+```cs
+[SkipOnCoreClr("https://github.com/dotnet/runtime/issues/45464", ~RuntimeConfiguration.Release)]
+```
+
+**Disable for combinations of options:**
+
+There are additional attribute signatures for combinations of these configurations, where all the conditions must be met:
+```cs
+SkipOnCoreClr(string reason, RuntimeConfiguration runtimeConfigurations, RuntimeTestModes testModes)
+SkipOnCoreClr(string reason, TestPlatforms testPlatforms, RuntimeConfiguration runtimeConfigurations)
+SkipOnCoreClr(string reason, TestPlatforms testPlatforms, RuntimeTestModes testMode)
+SkipOnCoreClr(string reason, TestPlatforms testPlatforms, RuntimeConfiguration runtimeConfigurations, RuntimeTestModes testModes)
+```
```

Comment:
it's probably out of scope, but nowhere better to put it -- `[Collection(nameof(DisableParallelization))]` anywhere. We put it on test classes to indicate that none of the tests in that class (which as usual run serially with respect to each other) may run concurrently with tests in another class. We use this for tests that use a lot of disk space or memory, or dominate all the cores, such that they are likely to disrupt any tests that run concurrently.